src/backend/distributed/commands/multi_copy.c

```diff
@@ -560,7 +561,11 @@ CopyToExistingShards(CopyStmt *copyStatement, QueryCompletionCompat *completionT
 
 		dest->receiveSlot(tupleTableSlot, dest);
 
-		processedRowCount += 1;
+		++processedRowCount;
+
+#if PG_VERSION_NUM >= PG_VERSION_14
```

Comment:
What about `CopyToNewShards`? I thought that we needed this logic there as well but with an `append` table I am able to see the progress. Do you know why?

Reply:
With append, `BeginCopyFrom` sets the total_bytes. However, `tuples_processed` should be set by us. It is a single line change, so pushed that as well

Reply:
And, bytes_processed is done in a lower level APIs that we already call: `NextCopyFrom ->NextCopyFromRawFields ->CopyReadLine ->CopyReadLineText ->CopyLoadInputBuf ->CopyLoadRawBuf `